### DISCUSSION TITLE
CodeQL causes qmake to segfault when building C++ project

### DISCUSSION THREAD

**Discussion by dail8859** [community/community#67835] [Code Security]
### Select Topic Area

Bug

### Body

I'm attempting to add CodeQL to my C++ Qt project. https://github.com/dail8859/NotepadNext/pull/445

It is set for analyzing "python" and "cpp". The job to analyze python builds the C++ application and _works correctly_, however when analyzing cpp it attempts to run the exact same `qmake` command to build the application but fails:

> /home/runner/work/_temp/0b34b594-7731-4bfe-b5d8-6142839a8614.sh: line 4:  8801 Segmentation fault      (core dumped) qmake NotepadNext.pro

The weird thing is there is _no_ differences in the python or cpp jobs (no conditional steps). So my only conclusion is github/codeql-action/init@v2 is doing something odd behind the scenes when the language is set to cpp.

Has anyone else ran across something like this?

**Comment by aibaars** [comment]
I've never seen that happen before. Could you re-run the failed job and enable debug logging? That should cause the codeql-action to upload a debug artifact. Hopefully the detailed logs (especially build-tracer.log) help explain what is going wrong.

See also: https://github.blog/changelog/2022-05-24-github-actions-re-run-jobs-with-debug-logging/

**Reply by dail8859** [reply]
I previously reran it with debugging enabled and was only looking for more console output. Wasn't aware it included a debug artifact as well. Thanks for the info! I reran the job in debug mode and I'm looking at the debug artifact now.

If anyone is interested: https://github.com/dail8859/NotepadNext/actions/runs/6261265831

**Reply by dail8859** [reply]
Dug through the logs but didn't see anything obvious myself :(

**Reply by aibaars** [reply]
I didn't see anything obvious myself either :( .

**Reply by aeisenberg** [reply]
This is some strange stuff. I have a few suggestions, but nothing seems like an obvious fix:

1. Your codeql workflow file does a lot of extra work between the initialization and analysis phase. Normally, this shouldn't be a problem, but every call is monitored by the codeql tracer (which is the parent process that helps build the codeql database). Can you try moving all of the non-build steps before the `codeql-action/init` init step? So, this would mean moving `jurplel/install-qt-action` and splitting the next run step so that the `apt-get` calls are performed earlier. It may be that these steps are interfering with the tracer.
2. Try calling `qmake` with the same arguments as in the [`build.yml` workflow](https://github.com/dail8859/NotepadNext/blob/master/.github/workflows/build.yml#L49).
3. I don't see any actions debugging information. Can you try re-running the workflow with the `debug` box selected?

**Reply by dail8859** [reply]
@aeisenberg Thanks for these suggestions! I had originally stuck close to the build.yml file but started to try different things to see if it was related to something I was doing. I also previously tried the specific version of Ubuntu that the build.yml is using and still the same results.

I made the suggested changes, but cpp still fails. I enabled debugging as well so there is an artifact attached.

https://github.com/dail8859/NotepadNext/actions/runs/6302919787

**Reply by aeisenberg** [reply]
Thanks for trying this out. I'll see if someone with more experience on the tracer has some thoughts.

**Comment by trevorsandy** [comment]
After upgrading my code base from Qt 5.15.2 to Qt 6.9.1, my C++ GitHub build is experiencing the behaviour described in this ticket.

~~~
+ qmake6 -nocache CONFIG+=release CONFIG-=debug_and_release CONFIG+=deb LPub3D.pro
builds/utilities/ci/github/linux-compile.sh: line 311: 13670 Segmentation fault      (core dumped) ${QMAKE_EXEC} -nocache CONFIG+=release CONFIG-=debug_and_release CONFIG+=${distropkg} LPub3D.pro
~~~

It's quite strange as qmake6 runs successfully in the job's 'Renderer' step but consistently fails with a segmentation fault in the cpp-language 'Compile' step.

I've re-run the job with debug logging but I did not find any meaningful information in the expanded build log.

You can analyze the build log [here:](https://github.com/trevorsandy/lpub3d-ci/actions/runs/16556297476)

Cheers,

**Reply by aeisenberg** [reply]
Can you please raise an issue in https://github.com/github/codeql/issues/new? Our first responder will be able to help you out.

**Reply by trevorsandy** [reply]
Very well - thank you for your assistance.

Cheers,
